src/languages/php.js

```diff
@@ -11,23 +11,105 @@ Category: common
  * @returns {LanguageDetail}
  * */
 export default function(hljs) {
+  const regex = hljs.regex;
+  const LABEL = '[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*' +
+    // negative look-ahead tries to avoid matching patterns that are not
+    // Perl at all like $ident$, @ident@, etc.
+    '(?![A-Za-z0-9])(?![$])';
+  const PASCAL_CASE_CLASS_NAME = regex.concat("[A-Z]", LABEL);
   const VARIABLE = {
-    className: 'variable',
-    begin: '\\$+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*' +
-      // negative look-ahead tries to avoid matching patterns that are not
-      // Perl at all like $ident$, @ident@, etc.
-      `(?![A-Za-z0-9])(?![$])`
+    scope: 'variable',
+    match: '\\$+' + LABEL,
+  };
+  const CONSTANT_CALL = regex.concat(LABEL, "\\b(?!\\()");
```

Comment:
What is the meaning of "CALL" here?  I thought it was a call/dispatch but then your regex seems to say "NOT" followed by `(` Rather than [must be followed by `(`], so I"m confused.

Reply:
PHP has:
- constant calling
`DateTimeImmutable::ISO8601;`
- static function invoke
`DateTimeImmutable::createFromMutable(new \DateTime());`


I do not want static function invoke to be highlighted as constant

Reply:
In many other grammars we'd:

- FIRST match the function invoke as `title.function`...
- then match for constants (which now will not require the look-ahead since those are already matched)

I'd rename this to `CONSTANT_REFERENCE` as I think that's more in line with the rest of the codebase... if "call" means something magical in PHP you can add a comment explaining the context further.

Reply:
So if part of the code will be matched as function invoke then another rule will not replace it with something else?
CONSTANT_REFERENCE looks good.

Reply:
> So if part of the code will be matched as function invoke then another rule will not replace it with something else?

Matching is sequential and "consumes" the content... two rules cannot match the exact same content (though keywords are matched INSIDE other content, so they can be different)... there are exceptions like `returnBegin` but we're slowly deprecating them and don't encourage their use... look-ahead is the only way to match without consuming.